File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
def solution(players, m, k):
    answer = 0
    servers = [[0] * 24 for r in range(2)]
    for i in range(24):
        need = (players[i] // m) - servers[0][i]
        if need > 0:
            servers[1][i] += need
            for j in range(k):
                if i+j >= 24:
                    break
                servers[0][i+j] += need
    for i in range(24):
        answer += servers[1][i]
    return answer

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
import java.util.*;

class Solution {
    public int solution(int[] players, int m, int k) {
        int[][] servers = new int[2][24];
        for(int i=0; i<24; i++){
            int need = players[i] / m - servers[0][i];
            if(need > 0){
                servers[1][i] += need;
                for(int j=i; j<i+k; j++){
                    if(j == 24){
                        break;
                    }
                    servers[0][j] += need;
                }
            }
        }
        int answer = 0;
        for(int i=0; i<24; i++){
            answer += servers[1][i];
        }
        return answer;
    }
}